### ISSUE TITLE
[BUG] _PmdArimaAdapter setting the wrong relative forecasts for frequency 10T

### ISSUE THREAD

**Issue by ParthivNaresh** [alan-turing-institute/sktime#805]
Currently while using a dataset that has a datetime index with a frequency of 10T (separated by 10 minutes), the ForecastingHorizon is setting the relative values to increments of 100. This leads to `n_periods` being set as 100 times the actual length `X.shape[0]` of the exogenous variable used in `predict`. This causes a `ValueError` to be raised by the Arima package `X array dims (n_rows) != n_periods`.

To reproduce:
1. Download the dataset `energy_data.csv` from [here](https://archive.ics.uci.edu/ml/machine-learning-databases/00374/).
2. Set the index as `date`.
3. Separate X and y.
4. Set `dates` as the index of `X` convert as a `datetime`.
5. Convert `dates` to PeriodIndex with the appropriate frequency.
6. Initialize `ForecastingHorizon`, followed by fitting and predicting with `AutoArima`.

```
data = pd.read_csv("../energydata_complete.csv")
data = data[:200]
data.set_index("date", drop=True, inplace=True)
X = data.copy()
y = X.pop("Appliances")

dates = pd.to_datetime(X.index)
freq = pd.infer_freq(dates)
dates = dates.to_period(freq=freq)

X.index = dates
y.index = dates

fh_ = ForecastingHorizon(y.index[150:200], is_relative=False)

a_clf = sktime_arima.AutoARIMA(start_p=2, start_q=2, max_p=5, max_q=5)
clf = a_clf.fit(X=X[:150], y=y[:150])
y_pred_sk = clf.predict(fh=fh_, X=X[150:200])
```

The expectation is that `n_periods = int(fh.to_relative(self.cutoff)[-1])` would set `n_periods` to 50 as there are 50 increments of datetime instances separated by 10 minutes being passed to `predict`. However `fh.to_relative(self.cutoff)` ends up producing this:

`Int64Index([ 100,  200,  300,  400,  500,  600,  700,  800,  900, 1000, 1100,
            1200, 1300, 1400, 1500, 1600, 1700, 1800, 1900, 2000, 2100, 2200,
            2300, 2400, 2500, 2600, 2700, 2800, 2900, 3000, 3100, 3200, 3300,
            3400, 3500, 3600, 3700, 3800, 3900, 4000, 4100, 4200, 4300, 4400,
            4500, 4600, 4700, 4800, 4900, 5000],
           dtype='int64')`

This however is the `ForecastingHorizon` that is used to product that relative index with a cutoff of `2016-01-12 17:50`:

`ForecastingHorizon(['2016-01-12 18:00', '2016-01-12 18:10', '2016-01-12 18:20',
             '2016-01-12 18:30', '2016-01-12 18:40', '2016-01-12 18:50',
             '2016-01-12 19:00', '2016-01-12 19:10', '2016-01-12 19:20',
             '2016-01-12 19:30', '2016-01-12 19:40', '2016-01-12 19:50',
             '2016-01-12 20:00', '2016-01-12 20:10', '2016-01-12 20:20',
             '2016-01-12 20:30', '2016-01-12 20:40', '2016-01-12 20:50',
             '2016-01-12 21:00', '2016-01-12 21:10', '2016-01-12 21:20',
             '2016-01-12 21:30', '2016-01-12 21:40', '2016-01-12 21:50',
             '2016-01-12 22:00', '2016-01-12 22:10', '2016-01-12 22:20',
             '2016-01-12 22:30', '2016-01-12 22:40', '2016-01-12 22:50',
             '2016-01-12 23:00', '2016-01-12 23:10', '2016-01-12 23:20',
             '2016-01-12 23:30', '2016-01-12 23:40', '2016-01-12 23:50',
             '2016-01-13 00:00', '2016-01-13 00:10', '2016-01-13 00:20',
             '2016-01-13 00:30', '2016-01-13 00:40', '2016-01-13 00:50',
             '2016-01-13 01:00', '2016-01-13 01:10', '2016-01-13 01:20',
             '2016-01-13 01:30', '2016-01-13 01:40', '2016-01-13 01:50',
             '2016-01-13 02:00', '2016-01-13 02:10'],
            dtype='period[10T]', name='date', freq='10T', is_relative=False)`

Versions
- Python (3.8.0)
- Sktime (0.5.3)
- pmdarima (1.8.0)

System:
    python: 3.8.0 | packaged by conda-forge | (default, Nov 22 2019, 19:11:19)  [Clang 9.0.0 (tags/RELEASE_900/final)]
executable: /Users/parthiv.naresh/miniconda3/envs/initial_env/bin/python
   machine: macOS-10.15.7-x86_64-i386-64bit

Python dependencies:
```
pip: 21.0.1
setuptools: 49.6.0.post20201009
sklearn: 0.24.0
sktime: 0.5.3
statsmodels: 0.12.2
numpy: 1.19.5
scipy: 1.5.3
Cython: 0.29.17
pandas: 1.1.5
matplotlib: 3.3.3
joblib: 1.0.0
numba: 0.52.0
pmdarima: 1.8.0
tsfresh: None
```

**Comment by mloning** [comment]
Thanks for raising the issue @ParthivNaresh! 

This is related to https://github.com/alan-turing-institute/sktime/issues/670

Would appreciate if you look into this. We should either fix it if possible, or raise an error.

**Comment by mloning** [comment]
@ParthivNaresh could you confirm if this is still an issue with the latest code on `main` after having merged #873?

**Comment by ParthivNaresh** [comment]
@mloning Sorry for the late response, I'm still seeing this error with `sktime==0.7.0`.

Using `1D` as the `freq`:
```
import numpy as np
import pandas as pd
from sktime.forecasting.arima import AutoARIMA
from sktime.forecasting.base import ForecastingHorizon

time_index = pd.date_range("January 1, 2021", periods=8, freq='1D')
X = pd.DataFrame(np.random.randint(0, 4, 24).reshape(8,3), columns=["First", "Second", "Third"], 
                 index=time_index)
y = pd.Series([1, 3, 2, 4, 5, 2, 3, 1], index=time_index)

fh_ = ForecastingHorizon(X.index[5:], is_relative=False)

a_clf = AutoARIMA(start_p=2, start_q=2, max_p=5, max_q=5)
clf = a_clf.fit(X=X[:5], y=y[:5])
y_pred_sk = clf.predict(fh=fh_, X=X[5:])

y_pred_sk
============================================================================

2021-01-06    3.778834
2021-01-07    7.151245
2021-01-08    5.387020
dtype: float64
```

Using `2D` as the `freq`:
```
import numpy as np
import pandas as pd
from sktime.forecasting.arima import AutoARIMA
from sktime.forecasting.base import ForecastingHorizon

time_index = pd.date_range("January 1, 2021", periods=8, freq='2D')
X = pd.DataFrame(np.random.randint(0, 4, 24).reshape(8,3), columns=["First", "Second", "Third"], 
                 index=time_index)
y = pd.Series([1, 3, 2, 4, 5, 2, 3, 1], index=time_index)

fh_ = ForecastingHorizon(X.index[5:], is_relative=False)

a_clf = AutoARIMA(start_p=2, start_q=2, max_p=5, max_q=5)
clf = a_clf.fit(X=X[:5], y=y[:5])
y_pred_sk = clf.predict(fh=fh_, X=X[5:])

y_pred_sk
============================================================================

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-37-8ef2c961eff7> in <module>
      3 a_clf = AutoARIMA(start_p=2, start_q=2, max_p=5, max_q=5)
      4 clf = a_clf.fit(X=X[:5], y=y[:5])
----> 5 y_pred_sk = clf.predict(fh=fh_, X=X[5:])

~/miniconda3/envs/main/lib/python3.8/site-packages/sktime/forecasting/base/_base.py in predict(self, fh, X, return_pred_int, alpha)
    156         # alpha = check_alpha(alpha)
    157 
--> 158         return self._predict(self.fh, X, return_pred_int=return_pred_int, alpha=alpha)
    159 
    160     def fit_predict(

~/miniconda3/envs/main/lib/python3.8/site-packages/sktime/forecasting/base/adapters/_pmdarima.py in _predict(self, fh, X, return_pred_int, alpha)
     56         # all values are out-of-sample
     57         if fh.is_all_out_of_sample(self.cutoff):
---> 58             return self._predict_fixed_cutoff(fh_oos, **kwargs)
     59 
     60         # all values are in-sample

~/miniconda3/envs/main/lib/python3.8/site-packages/sktime/forecasting/base/adapters/_pmdarima.py in _predict_fixed_cutoff(self, fh, X, return_pred_int, alpha)
    104         # make prediction
    105         n_periods = int(fh.to_relative(self.cutoff)[-1])
--> 106         result = self._forecaster.predict(
    107             n_periods=n_periods,
    108             X=X,

~/miniconda3/envs/main/lib/python3.8/site-packages/pmdarima/utils/metaestimators.py in <lambda>(*args, **kwargs)
     51 
     52         # lambda, but not partial, allows help() to work with update_wrapper
---> 53         out = (lambda *args, **kwargs: self.fn(obj, *args, **kwargs))
     54         # update the docstring of the returned function
     55         update_wrapper(out, self.fn)

~/miniconda3/envs/main/lib/python3.8/site-packages/pmdarima/arima/auto.py in predict(self, n_periods, X, return_conf_int, alpha, **kwargs)
    255         # Temporary shim until we remove `exogenous` support completely
    256         X, _ = pm_compat.get_X(X, **kwargs)
--> 257         return self.model_.predict(
    258             n_periods=n_periods,
    259             X=X,

~/miniconda3/envs/main/lib/python3.8/site-packages/pmdarima/arima/arima.py in predict(self, n_periods, X, return_conf_int, alpha, **kwargs)
    668         X = self._check_exog(X)  # type: np.ndarray
    669         if X is not None and X.shape[0] != n_periods:
--> 670             raise ValueError('X array dims (n_rows) != n_periods')
    671 
    672         # f = self.arima_res_.forecast(steps=n_periods, exog=X)

ValueError: X array dims (n_rows) != n_periods
```

**Comment by mloning** [comment]
Thanks @ParthivNaresh - this is something we need to further investigate then!

**Comment by khrapovs** [comment]
Here is the root cause of this bug:
```
index = pd.period_range("January 1, 2021", periods=3, freq='2D')
index - index[0]

Index([<0 * Days>, <4 * Days>, <8 * Days>], dtype='object')

[val - index[0] for val in index]

[<0 * Days>, <2 * Days>, <4 * Days>]
```
Can somebody understand WTF is going on here???

**Comment by khrapovs** [comment]
Same creepiness is happening in https://github.com/alan-turing-institute/sktime/issues/1935

**Comment by fkiraly** [comment]
>Can somebody understand [blank] is going on here???

The root cause seems to be a `pandas` bug, as identified in #1935.

**Comment by khrapovs** [comment]
With this little change https://github.com/alan-turing-institute/sktime/pull/1981 both of the examples in https://github.com/alan-turing-institute/sktime/issues/805#issuecomment-891848228 run as expected.

**Comment by khrapovs** [comment]
Now it makes sense:
```
index = pd.period_range("January 1, 2021", periods=3, freq='2D').to_timestamp()
print(index - index[0])
print([val - index[0] for val in index])

TimedeltaIndex(['0 days', '2 days', '4 days'], dtype='timedelta64[ns]', freq=None)
[Timedelta('0 days 00:00:00'), Timedelta('2 days 00:00:00'), Timedelta('4 days 00:00:00')]
```

**Comment by fkiraly** [comment]
yes, but a lot of other failures pop up... ☹️

**Comment by fkiraly** [comment]
This should be fixed by #1981.
